include/grpc/event_engine/event_engine.h

```diff
@@ -481,7 +481,7 @@ class EventEngine : public std::enable_shared_from_this<EventEngine>,
 /// created, applications must set a custom EventEngine factory method *before*
 /// grpc is initialized.
 void SetEventEngineFactory(
```

Comment:
Once callers are migrated to this new API, we're going to remove this method and the next one, right?  If so, please add TODOs.